fw/apm.c

```diff
@@ -1327,30 +1326,45 @@ __tfw_apm_crc32_calc(TfwMsgIter *it, TfwStr *chunk , struct sk_buff *skb_head,
 }
 
 bool
-tfw_apm_hm_srv_alive(int status, TfwStr *body, struct sk_buff *skb_head,
-		     void *apmref)
+tfw_apm_hm_srv_alive(TfwHttpResp *resp, TfwServer *srv)
 {
-	TfwApmHM *hm = READ_ONCE(((TfwApmRef *)apmref)->hmctl.hm);
+	int status = resp->status;
+	TfwStr *body = &resp->body;
+
+	TfwApmHM *hm = READ_ONCE(((TfwApmRef *)(srv->apmref))->hmctl.hm);
+
 	u32 crc32 = 0;
 	TfwMsgIter it;
 	TfwStr chunk = {0};
 
 	BUG_ON(!hm);
+       /*
+       * If hm->codes exists
+       * and 'status' has not in hm->codes.
+       * return NOT ALIVE
+       */
 	if (hm->codes && !test_bit(HTTP_CODE_BIT_NUM(status), hm->codes)) {
 		T_WARN_NL("Response for health monitor '%s': status '%d' "
 			  "mismatched\n", hm->name, status);
 		return false;
 	}
-
-	if (!body->len) {
+	/*
+	 * Check CRC for zero body
+	 * and return ALIVE if CRC also zero,
+	 * go to process CRC error if not
+	 * and continue processing if body exists.
+	 */
+	if (body->len == 0) {
 		if (hm->crc32 != 0)
 			goto crc_err;
 		return true;
 	}
 
-	if (unlikely(tfw_body_iter_init(&it, &chunk, body->data, body->skb,
-					skb_head)))
-	{
+	/*
+	 * If body is invalid return NOT ALIVE
+	 */
```

Comment:
I believe this is an unnecessary comment as it simply repeats what is already clearly expressed in the code.